src/Forward.cpp

```diff
@@ -0,0 +1,77 @@
+#include <iodrivers_base/Forward.hpp>
+#include <iodrivers_base/Driver.hpp>
+#include <thread>
+#include <vector>
+
+using namespace std;
+using namespace iodrivers_base;
+
+struct QuitGuard
+{
+    bool& quitFlag;
+    Driver& driver1;
+    Driver& driver2;
+
+    ~QuitGuard() {
+        quitFlag = true;
+    }
+};
+
+void iodrivers_base::forward(bool raw_mode,
+                Driver& driver1, Driver& driver2,
+                base::Time timeout1,
+                base::Time timeout2,
+                size_t const buffer_size)
+{
+    vector<uint8_t> buffer(buffer_size);
+    int fd1 = driver1.getFileDescriptor();
+    int fd2 = driver2.getFileDescriptor();
+    int select_nfd = std::max(fd1, fd2) + 1;
+
+
+    typedef int (Driver::*ReadMode)(uint8_t*, int, base::Time const&);
+    ReadMode readMode = raw_mode ? static_cast<ReadMode>(&Driver::readRaw) :
+                                   static_cast<ReadMode>(&Driver::readPacket);
+
+    while (!driver1.eof() && !driver2.eof()) {
+        fd_set set;
+        FD_ZERO(&set);
+        FD_SET(fd1, &set);
+        FD_SET(fd2, &set);
+
+        timeval timeout_spec = {
+            static_cast<time_t>(10),
+            suseconds_t(0)
+        };
+
+        int ret = select(select_nfd, &set, NULL, NULL, &timeout_spec);
```

Comment:
poll() provides a nicer interface in my opinion.. No need to keep resetting the fd_set at every iteration..

Reply:
For a single FD, it really makes no difference IMO. Since `iodrivers_base` uses select already, I stuck to that.